src/inputs/TextFieldBase.tsx

```diff
@@ -243,8 +247,7 @@ export function TextFieldBase<X extends Only<TextFieldXss, X>>(props: TextFieldB
                 {...(errorMsg ? { "aria-errormessage": errorMessageId } : {})}
                 ref={fieldRef as any}
                 rows={multiline ? 1 : undefined}
-                // Make the input field readOnly if the field explicitly sets it to `true`
-                readOnly={inputProps.readOnly}
+                readOnly={preventEdit}
```

Comment:
Question: Hm, we don't need to check `inputProps.readOnly` at all anymore? I would have expected an `||` or something...

Reply:
We use `inputProps.readOnly` to conditionally show our custom readOnly view. It'll never be true in this code block, so is safe to not consider it here.